Python/ceval_macros.h

```diff
@@ -62,13 +62,17 @@
 #ifdef Py_STATS
 #define INSTRUCTION_START(op) \
     do { \
-        frame->prev_instr = next_instr++; \
+        frame->instr_ptr = next_instr++; \
         OPCODE_EXE_INC(op); \
         if (_Py_stats) _Py_stats->opcode_stats[lastopcode].pair_count[op]++; \
         lastopcode = op; \
     } while (0)
 #else
-#define INSTRUCTION_START(op) (frame->prev_instr = next_instr++)
+#define INSTRUCTION_START(op) \
+    do { \
+        frame->instr_ptr = next_instr++; \
+        frame->next_instr_offset = 0; \
```

Comment:
> Why is this necessary? 

For line monitoring, to be able to find the previous instruction.

Reply:
Line monitoring was able to find the previous instruction before. Why can't it now?
I would assume that at the very start of an instruction (immediately after the jump, whether `switch` or `goto`) that `next_instr` points to the instructions about to be executed and `instr_ptr` points to the instruction just executed.
Is that not the case, and if not, what do they point to?

Reply:
It's not the case because returns apply the `next_instr_offset` to `instr_ptr`. I think they should leave `instr_ptr` alone and just update `next_instr`, but where we jump to `resume_frame:`, the next thing that happens is `next_instr = frame->instr_ptr`, so this expects `instr_ptr` to have been bumped.

Reply:
I pushed this change, and I think there is just one test failing, related to entering an executor on a jump with `EXTENDED_ARG`.

Reply:
I can also rename next_instr_offset back to return_offset now, because YIELD is no longer using it.